File: TheaterSuggestions/Java/ScreeningSeatingArrangementSuggestions/src/main/java/org/weaveit/seatingplacesuggestions/SeatingOptionIsSuggested.java
package org.weaveit.seatingplacesuggestions;

import java.util.List;

public record SeatingOptionIsSuggested(
        int partyRequested,
        PricingCategory pricingCategory,
        List<SeatingPlace> seats
) implements SeatingOption {

    public SeatingOptionIsSuggested {
        seats = List.copyOf(seats);
    }

    @Override
    public boolean matchExpectation() {
        return seats.size() == partyRequested;
    }
}

File: TheaterSuggestions/Kotlin/ScreeningSeatingArrangementSuggestions/src/main/kotlin/org/weaveit/seatingplacesuggestions/SeatingOptionIsSuggested.kt
package org.weaveit.seatingplacesuggestions

data class SeatingOptionIsSuggested private constructor(
    override val partyRequested: Int,
    override val pricingCategory: PricingCategory,
    private val _seats: List<SeatingPlace>
) : SeatingOption {

    override fun seats(): List<SeatingPlace> = _seats

    companion object {
        operator fun invoke(
            partyRequested: Int,
            pricingCategory: PricingCategory,
            seats: List<SeatingPlace>
        ): SeatingOptionIsSuggested =
            SeatingOptionIsSuggested(partyRequested, pricingCategory, seats.toList())
    }

    override fun matchExpectation(): Boolean {
        return _seats.size == partyRequested
    }
}
